### PR TITLE
Adjust error logging in Jetpack Connection Pilot

### PR DESCRIPTION
**Pull request by WPprodigy**
Mostly just altering the error logging and cleaning up the way errors are passed around so that we'll have a better idea of the full execution path when something goes wrong.

Also removed some unnecessary filtering, and instead just performed the actions inline.

### CODE REVIEW

File: vip-jetpack/connection-pilot/class-jetpack-connection-controls.php [hunk 1]
```diff
@@ -288,6 +302,8 @@ private static function maybe_create_user() {
 	 * This helps prevent cache issues for times where the database was directly updated.
 	 */
 	private static function refresh_options_cache() {
+		wp_cache_flush_runtime();
```
Comment (WPprodigy) [inline vip-jetpack/connection-pilot/class-jetpack-connection-controls.php:305]: Small addition just to help ensure the request doesn't carry over some other stale object cache values that may have been updated via the roundtrips with Jetpack.

File: vip-jetpack/connection-pilot/class-jetpack-connection-controls.php [hunk 2]
```diff
@@ -168,31 +167,31 @@ public static function connect_site( bool $skip_connection_tests = false, bool $
...
-
-			if ( is_akismet_key_invalid() ) {
-				$original_user = wp_get_current_user();
-				// Getting wpcomvip user, since it's the owner of the Jetpack connection
-				$vip_user = get_user_by( 'login', 'wpcomvip' );
-				if ( ! $original_user || ! $vip_user ) {
-					return false;
-				}
+		if ( ! class_exists( 'Akismet_Admin' ) || ! method_exists( 'Akismet_Admin', 'connect_jetpack_user' ) || ! function_exists( 'is_akismet_key_invalid' ) ) {
+			return new WP_Error( 'jp-cxn-pilot-akismet-dependencies-missing', 'Akismet is missing required functions/methods to perform the connection.' );
```
Comment (rinatkhaziev) [inline vip-jetpack/connection-pilot/class-jetpack-connection-controls.php:174]: This doesn't satisfy `bool` return type.

File: vip-jetpack/connection-pilot/class-jetpack-connection-controls.php [hunk 3]
```diff
@@ -168,31 +167,31 @@ public static function connect_site( bool $skip_connection_tests = false, bool $
...
 
-				return $result;
-			}
+		// Put the machine user in charge of things.
+		wp_set_current_user( $user->ID );
 
-			return true;
+		$result = \Akismet_Admin::connect_jetpack_user();
+		if ( ! $result ) {
+			return new WP_Error( 'jp-cxn-pilot-akismet-connection-failed', 'Akismet could not be connected.' );
```
Comment (rinatkhaziev) [inline vip-jetpack/connection-pilot/class-jetpack-connection-controls.php:191]: same here

**Review by rinatkhaziev — CHANGES_REQUESTED** [review]
I think there's possibility of fatals now because return type might not match in connect_akismet - left inline notes.

File: vip-jetpack/connection-pilot/class-jetpack-connection-controls.php (hunk 2 above)
Comment (WPprodigy) [reply vip-jetpack/connection-pilot/class-jetpack-connection-controls.php:174]: nice catch! fixed

**Pull request merged**